src/datajudge/db_access.py

```diff
@@ -24,10 +25,46 @@ def is_snowflake(engine: sa.engine.Engine) -> bool:
     return engine.name == "snowflake"
 
 
+def is_bigquery(engine: sa.engine.Engine) -> bool:
+    return engine.name == "bigquery"
+
+
 def get_table_columns(table, column_names):
     return [table.c[column_name] for column_name in column_names]
 
 
+def apply_patches(engine: sa.engine.Engine):
```

Comment:
Interesting.

My only concern is that this function has to be called manually for people using `bigquery`. In the tests, you're taking care of that, but users won't be directly aware.

For now, since we don't have a section with backend-specific comments, you could write in the documentation at the bottom of [this section](https://datajugde.readthedocs.io/en/latest/getting_started.html#creating-a-specification) what to do for `bigquery`.

Furthermore, our `pytest` "integration" could also handle it automatically by adding to [this function](https://github.com/Quantco/datajudge/blob/8ab90bdb291db44534d4447df85a00fc4025261a/src/datajudge/pytest_integration.py#L23) an implicit call to the `apply_patch` function before the `datajudge_engine` fixture is used.

What's your take @kklein?

Reply:
I think you raise great points @ivergara .

> but users won't be directly aware.

As you also point out, users will likely start off with some code samples from the documentation - indicating how the engine object has to be made accessible for tests. I figure that adding a comment in said documentation to call this function should be a good start. 

> Furthermore, our pytest "integration" could also handle it automatically by adding to [this function](https://github.com/Quantco/datajudge/blob/8ab90bdb291db44534d4447df85a00fc4025261a/src/datajudge/pytest_integration.py#L23) an implicit call to the apply_patch function before the datajudge_engine fixture is used.

I don't have an opinion on which of both options to pursue.

I think in general we might want to keep an eye on this topic (and potentially open issues at https://github.com/googleapis/python-bigquery-sqlalchemy) and think of ways to make the usage easiest for an end user. Yet, the suggested approach looks perfectly fine to me to get started with BigQuery. :)

Reply:
Very good point, I missed that. I like @ivergara 's suggestion to add the patch before using the datajudge_engine fixture. Alternatively, one could have a second "engine" fixture which return a patched datajudge_engine one. But since the patch should only be necessary until sqlalchemy fix the issues, I wouldn't go for it.